test/fish_prompt_test.fish

```diff
@@ -23,44 +23,57 @@ function assert \
   end
 end
 
+function os_prompt_path_prefix
```

Comment:
One thing I was thinking, maybe we should also prefix the function names inside here? Since terminal is weird and it all taints everything.

Reply:
@lsunsi, yup, totally agreed.

Reply:
I mean, if it set the function outside the scope it runs, we might have to namespace it. But like the variable, it doesn't get set.

Yeah, I think fisherman is smart.